apps/passport-client/components/screens/MultiChoiceScanScreen.tsx

```diff
@@ -0,0 +1,219 @@
+import { useCallback, useState } from "react";
+import { QrReader } from "react-qr-reader";
+import { useNavigate } from "react-router-dom";
+import { SingleValue } from "react-select";
+import styled from "styled-components";
+import { useDispatch, useLaserScannerKeystrokeInput } from "../../src/appHooks";
+import { loadUsingLaserScanner } from "../../src/localstorage";
+import { maybeRedirect } from "../../src/util";
+import { H5, Spacer, TextCenter } from "../core";
+import { ScanditScanner } from "../core/Scandit";
+import Scanner from "../core/Scanner";
+import { AppContainer } from "../shared/AppContainer";
+import { IndicateIfOffline } from "../shared/IndicateIfOffline";
+import Select from "../shared/Select";
+import {
+  Back,
+  Home
+} from "./ScannedTicketScreens/PodboxScannedTicketScreen/PodboxScannedTicketScreen";
+
+const ButtonsContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: row;
+  gap: 8px;
+  margin-bottom: 16px;
+
+  button {
+    flex-grow: 1;
+  }
+`;
+
+// Scan a PCD QR code, then go to /verify to verify and display the proof.
+export function MultiChoiceScanScreen(): JSX.Element {
+  const usingLaserScanner = loadUsingLaserScanner();
+  useLaserScannerKeystrokeInput();
+  const nav = useNavigate();
+  const dispatch = useDispatch();
+
+  const [scanner, setScanner] = useState<
+    "strich" | "react-qr-reader" | "scandit"
+    // @ts-expect-error this is temporary code so doesn't need to type-check
+  >(localStorage.getItem("preferred-scanner") ?? "scandit");
+
+  type Option = {
+    id: "strich" | "react-qr-reader" | "scandit";
+    label: string;
+  };
+
+  const onChange = useCallback(
+    (option: SingleValue<Option>) => {
+      if (option) {
+        setScanner(option.id);
+        localStorage.setItem("preferred-scanner", option.id);
+      }
+    },
+    [setScanner]
+  );
+
+  const options: Option[] = [
+    { id: "scandit", label: "Scandit" },
+    { id: "strich", label: "Strich" },
+    { id: "react-qr-reader", label: "React-QR-Reader" }
+  ];
+
+  const onResult = useCallback(
+    (result: string): void => {
+      console.log(`Got result, considering redirect`, result);
+      const newLoc = maybeRedirect(result);
+      if (newLoc) {
+        nav(newLoc);
+      } else {
+        dispatch({
+          type: "error",
+          error: {
+            title: "Not a Zupass QR code",
+            message:
+              "The QR code you scanned is not a Zupass QR code. Make sure the QR code you're scanning comes from the Zupass app.",
+            dismissToCurrentPage: true
+          }
+        });
+      }
+    },
+    [dispatch, nav]
+  );
+
+  const onQrReaderResult = useCallback(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
does this library have typescript types? if not, and we choose to use it, we should create our own `.d.ts` file